File: mycollab-web/src/main/java/com/esofthead/mycollab/module/user/events/UserEvent.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.esofthead.mycollab.module.user.events;

import com.esofthead.mycollab.eventmanager.ApplicationEvent;

/**
 *
 * @author haiphucnguyen
 */
@SuppressWarnings("serial")
public class UserEvent {

    
	public static class PlainLogin extends ApplicationEvent {

        public PlainLogin(Object source, Object data) {
            super(source, data);
        }
    }
    
    public static class Search extends ApplicationEvent {

        public Search(Object source, Object data) {
            super(source, data);
        }
    }
    
    public static class GotoList extends ApplicationEvent {

        public GotoList(Object source, Object data) {
            super(source, data);
        }
    }
    
    public static class GotoAdd extends ApplicationEvent {

        public GotoAdd(Object source, Object data) {
            super(source, data);
        }
    }

    public static class GotoRead extends ApplicationEvent {

        public GotoRead(Object source, Object data) {
            super(source, data);
        }
    }

    public static class GotoEdit extends ApplicationEvent {

        public GotoEdit(Object source, Object data) {
            super(source, data);
        }
    }
}

File: mycollab-web/src/main/scala/com/esofthead/mycollab/module/user/events/UserEvent.scala
package com.esofthead.mycollab.module.user.events

import com.esofthead.mycollab.eventmanager.ApplicationEvent

import scala.beans.BeanProperty

/**
 * @author MyCollab Ltd
 * @version 5.1.0
 */
object UserEvent {

    class PlainLogin(@BeanProperty val username: String, @BeanProperty val password: String,
                     @BeanProperty val isRememberMe: Boolean) extends Serializable {}

    class Search(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

    class GotoList(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

    class GotoAdd(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

    class GotoRead(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

    class GotoEdit(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

}
